packages/node_modules/nav-frontend-popover/src/popover.tsx

```diff
@@ -0,0 +1,221 @@
+
+import { EventThrottler, keyCodes, getViewportDimensions, getScrollParents } from 'nav-frontend-js-utils';
+
+import PopoverBase, { PopoverBaseProps, PopoverPosisjonShape, PopoverOrientering } from './popover-base';
+
+const cls = (state, props) => classnames('popover--controlled', {
+    'popover--hidden': !state.åpen
+});
+
+export interface PopoverProps extends PopoverBaseProps {
+    /**
+     * Elementet som popover er forankret til. Dette elementet blir brukt til å posisjonere popover.
+     */
+    ankerEl?: HTMLElement;
+    /**
+     * Callback når popover åpnes.
+     */
+    onOpen?: () => void;
+    /**
+     * Callback når popover ber om å lukkes.
+     */
+    onClose: () => void;
+    /**
+     * Avstand til anker element i pixler. Default er `16px` (`1rem`) med pil, eller `0` uten pil.
+     */
+    avstandTilAnker?: number;
+    /**
+     * Bestemmer om Popover automatisk skal få fokus når den vises.
+     */
+    autoFokus?: boolean;
+}
+
+interface PopoverState {
+    åpen: boolean;
+    posisjon?: PopoverPosisjonShape;
+}
+
+class Popover extends React.Component<PopoverProps, PopoverState> {
+    static defaultProps: Partial<PopoverProps> = {
+        autoFokus: true
+    };
+
+    private popoverRef = React.createRef<PopoverBase>();
+    private popoverEl;
+    private cachedMeasurements;
+    private listenToScrollOn;
+    private scrollParents: (HTMLElement|Window)[] = [];
+
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            åpen: (this.props.ankerEl !== undefined)
+        };
+
+        window.addEventListener('click', this.handleClick);
+        window.addEventListener('keydown', this.handleKeydown);
+    }
+
+    componentDidMount() {
+        this.popoverEl = ReactDOM.findDOMNode(this.popoverRef.current);
+        this.scrollParents = getScrollParents(this.popoverEl);
+        this.scrollParents.forEach((scrollParent) => scrollParent.addEventListener('scroll', this.handleScroll));
+    }
+
+    componentWillUnmount() {
+        window.removeEventListener('click', this.handleClick);
+        window.removeEventListener('keydown', this.handleKeydown);
+        this.scrollParents.forEach((scrollParent) => scrollParent.removeEventListener('scroll', this.handleScroll));
+    }
+
+    componentWillReceiveProps(nextProps) {
+        if (!this.props.ankerEl && nextProps.ankerEl) {
+            this.setState({
+                åpen: true
+            }, () => { // tslint:disable-line:align
+                this.updatePosition(nextProps);
+                if (this.props.autoFokus) this.popoverEl.focus();
+            });
+        } else if (this.props.ankerEl && !nextProps.ankerEl) {
+            this.setState({
+                åpen: false
+            });
+        }
+    }
+
+    checkFocus = () => {
+        window.setTimeout(() => {
+            const focusElement = document.activeElement;
+            const popoverEl = ReactDOM.findDOMNode(this.popoverRef.current);
```

Comment:
findDOMNode ligger an til å bli deprecated tror jeg

Reply:
Hm, jau. Betyr det at vi egentlig må gå gjennom alle pakkene våre og legge til "ref forwarding"?

Reply:
Hm, [virker nesten](https://github.com/storybookjs/storybook/issues/5974) som at vi må beholde exports som vi har i dag for at `react-docgen-typescript` fortsatt skal fungere. I så fall må vi gjøre noa ala:

```jsx
const KnappForwardRef = React.forwardRef<any, any>(
    (props, ref: any) => <Knapp innerRef={ref} {...props} />
);
export { KnappForwardRef };
export default Knapp;
```

Og da faller vel mye av vitsen bort?